src/include/processor/result/factorized_table.h

```diff
@@ -32,56 +32,51 @@ struct BlockAppendingInfo {
 // released when this struct goes out of scope.
 class DataBlock {
 public:
-    explicit DataBlock(storage::MemoryManager* memoryManager)
-        : numTuples{0}, memoryManager{memoryManager} {
-        block = memoryManager->allocateBuffer(true /* initializeToZero */);
-        freeSize = block->allocator->getPageSize();
+    DataBlock(storage::MemoryManager* mm, uint64_t size)
+        : numTuples{0}, totalSize{size}, freeSize{size} {
+        block = mm->allocateBuffer(true /* initializeToZero */, size);
     }
 
-    DataBlock(DataBlock&& other) = default;
-
-    inline uint8_t* getData() const { return block->buffer; }
-    inline void resetNumTuplesAndFreeSize() {
-        freeSize = common::BufferPoolConstants::PAGE_256KB_SIZE;
+    uint8_t* getData() const { return block->buffer; }
+    uint8_t* getWritableData() const { return block->buffer + totalSize - freeSize; }
```

Comment:
I suggest 
1. we remove the constant keyword, since the caller will write/modify the memory address returned by `getWritableData`.
2. Or we rename it to getDataUnsafe()?